frontend/src/pages/projects/screens/detail/data-connections/__tests__/utils.spec.ts

```diff
@@ -0,0 +1,241 @@
+  });
+
+  it('should return data connection description when dataconnection is an AWS data connection', () => {
+    dataConnection.data.metadata.annotations = { 'openshift.io/description': 'description' };
+    const dataConnectionDescription = getDataConnectionDescription(
+      createDataConnection(DataConnectionType.AWS),
+    );
+    expect(dataConnectionDescription).toBe('description');
+  });
+});
+
+describe('getDataConnectionType', () => {
+  it('should return connection type when data connection is an AWS data connection', () => {
+    const connectionType = getDataConnectionType(createDataConnection(DataConnectionType.AWS));
+    expect(() => {
+      connectionType;
+    }).not.toThrow('Invalid data connection type');
+    expect(connectionType).not.toBe(null);
+  });
+
+  it('should return error when dataconnection is not an AWS data connection', () => {
+    expect(() => {
+      getDataConnectionType(createDataConnection({} as DataConnectionType));
+    }).toThrow('Invalid data connection type');
+  });
+});
+
+describe('getDataConnectionProvider', () => {
+  it('should return data connection provider when dataconnection is AWS data connection', () => {
+    const connectionProvider = getDataConnectionProvider(
+      createDataConnection(DataConnectionType.AWS),
+    );
+    expect(connectionProvider).toBe('AWS S3');
+  });
+
+  it('should return error when dataconnection is not AWS data connection', () => {
+    expect(() => {
+      getDataConnectionProvider(createDataConnection({} as DataConnectionType));
+    }).toThrow('Invalid data connection type');
+  });
+});
+
+describe('deleteDataConnection', () => {
+  it('should return Success when dataconnection is AWS data connection', async () => {
+    const mockK8sStatus = {
+      code: 200,
+      message: '',
+      reason: 'string',
+      status: 'Success',
+    };
+    deletesecret.mockResolvedValue(mockK8sStatus);
+    const deleteConnection = await deleteDataConnection(
+      createDataConnection(DataConnectionType.AWS),
+    );
+    expect(deleteConnection.status).toBe('Success');
+  });
+
+  it('should return Failure when dataconnection is AWS data connection', async () => {
+    const mockK8sStatus = {
+      code: 404,
+      message: '',
+      reason: 'string',
+      status: 'Failure',
+    };
+    deletesecret.mockResolvedValue(mockK8sStatus);
+    const deleteConnection = await deleteDataConnection(
+      createDataConnection(DataConnectionType.AWS),
+    );
+    expect(deleteConnection.status).toBe('Failure');
+  });
+
+  it('should return error when dataconnection is not AWS data connection', () => {
+    expect(() => {
+      deleteDataConnection(createDataConnection({} as DataConnectionType));
+    }).toThrow('Invalid data connection type');
+  });
+});
+
+describe('convertAWSSecretData', () => {
+  it('should return converted/decoded secret data', () => {
+    dataConnection.data.data = {
```

Comment:
Better off creating the mock in this test and modifying it rather than modifying a shared object.